pkg/printers/internalversion/describe.go

```diff
@@ -2633,13 +2633,18 @@ func printReplicaSetsByLabels(matchingRSs []*versionedextension.ReplicaSet) stri
 	return list
 }
 
-func getPodStatusForController(c coreclient.PodInterface, selector labels.Selector) (running, waiting, succeeded, failed int, err error) {
+func getPodStatusForController(c coreclient.PodInterface, selector labels.Selector, uid types.UID) (running, waiting, succeeded, failed int, err error) {
 	options := metav1.ListOptions{LabelSelector: selector.String()}
 	rcPods, err := c.List(options)
 	if err != nil {
 		return
 	}
 	for _, pod := range rcPods.Items {
+		controllerRef := controller.GetControllerOf(&pod)
+		// Skip pods that are orphans or owned by other controllers.
+		if controllerRef != nil && controllerRef.UID != uid {
```

Comment:
This still doesn't skip orphans (`controllerRef = nil`).

It should be:

```go
if controllerRef == nil || controllerRef.UID != uid {
  continue
}
```

Please also add an orphan to the unit test to verify this works.